config/routes.rb

```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
-  root "articles#index"
-
-  get "/articles", to: "articles#index"
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  get '/users', to: 'users#index', as: 'users' 
+  get '/users/:id', to: 'users#show', as: 'user'
```

Comment:
- [**OPTIONAL**] You did an excellent job with the routes 👏. However, I recommend that you use `resources` to make your work easier and less typing. It simplifies the addition of nested routes. You can visit [this link](https://guides.rubyonrails.org/routing.html) for more information. 🤗
So you can refactor your routes as follow:
```  
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
 end

```